tools/yulPhaser/Chromosome.cpp

```diff
@@ -72,5 +72,5 @@ string const& Chromosome::randomOptimisationStep()
 {
 	static vector<string> stepNames = allStepNames();
 
-	return stepNames[SimulationRNG::uniformInt(0, stepNames.size() - 1)];
+	return stepNames[SimulationRNG::uniformInt(0, stepNames.size()) - 1];
```

Comment:
Is this a conversion warning fix or a bug fix? :wink:

Reply:
Also wouldn't this result in `0 - 1` when the rng gives 0?

Reply:
This is wrong definitely (in process of changing conversions back and fourth it ended up with this), @cameel wanted to take over this part, so he will fix it I guess :)

Reply:
Yeah, looks like a bug. I'll fix it.
BTW, the PR is in the draft state now. I'll mark it ready for review when I'm done with it.

Reply:
Fixed.